### ISSUE TITLE
safeitemrootname

### ISSUE THREAD

**Issue by DaanV2** [MicrosoftDocs/visualstudio-docs#1348]
safeitemrootname is not mentioned but used in the standard templates

---
#### Documentdetails

⚠ *Dit gedeelte niet bewerken. Het is vereist om problemen te koppelen tussen docs.microsoft.com en GitHub.*

* ID: 5f4a0078-1a87-6e3a-791e-d3068fb7032c
* Version Independent ID: 1d2d976d-2698-88d6-fac4-4d34e3855e2a
* Content: [Visual Studio project and item template parameters - Visual Studio](https://docs.microsoft.com/nl-nl/visualstudio/ide/template-parameters)
* Content Source: [docs/ide/template-parameters.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/ide/template-parameters.md)
* Product: **visual-studio-dev15**
* GitHub Login: @gewarren
* Microsoft Alias: **gewarren**

**Comment by gewarren** [comment]
@DaanV2 Thanks for bringing this to our attention.

**Comment by gewarren** [comment]
@DaanV2 Is safeitemrootname the same as safeitemname without the file extension?

**Comment by DaanV2** [comment]
I have both of them in an item template and got the following:

It takes the file name of the item and replaces all dangerous characters away.

Example: in this template, I create a file through the following:
The name of the item that I filled in. 
```
TestClass.cs```

The file that is created through the MyTemplate.vstemplate:
```xml
<ProjectItem TargetFileName="$fileinputname$ - Initialize.cs" ReplaceParameters="true">Class - Initialize.cs</ProjectItem>```

Creates a file called:
``` 
TestClass - Initialize.cs```

both $safeitemrootname$ and $safeitemname$ got renamed to:
TestClass___Initialize

I tested this also on an item template that generates 3 files and had each of the 3 files add in both safeitemrootname and safeitemname. 
It seems that it is always converted to the same.
